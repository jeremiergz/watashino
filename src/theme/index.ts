import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    html, body {
        background-color: ${({ theme }: { theme: Theme }) => theme.background};
        height: 100%;
        width: 100%;
        margin: 0;
        padding: 0;
        transition: background-color 250ms ease;
    }
    body {
        color: ${({ theme }: { theme: Theme }) => theme.colors.text};
        font-family: ${({ theme }: { theme: Theme }) => theme.fonts.main};
        font-weight: ${({ theme }: { theme: Theme }) => theme.fontWeights.regular};
    }
    #___gatsby {
        position: relative;
        min-height: 100%;
    }
    /* Format Markdown images loaded by gatsby-remark-images */
    .gatsby-resp-image-wrapper {
        .gatsby-resp-image-background-image, .gatsby-resp-image-image {
            border-radius: 8px;
        }
    }
    .grvsc-container {
      text-align: left;
    }
`;

const breakpoints = ['320px', '375px', '425px', '768px', '1024px', '1200px', '1440px', '2560px'] as const;
Object.defineProperties(breakpoints, {
  mobileS: { value: breakpoints[0] },
  mobileM: { value: breakpoints[1] },
  mobileL: { value: breakpoints[2] },
  tablet: { value: breakpoints[3] },
  laptopS: { value: breakpoints[4] },
  laptopM: { value: breakpoints[5] },
  laptopL: { value: breakpoints[6] },
  desktop: { value: breakpoints[7] },
});

const colors = {
  black: '#212121',
  dark: '#424242',
  error: '#f44336',
  gray: '#757575',
  info: '#2196f3',
  lightgray: '#e0e0e0',
  primary: '#31859a',
  secondary: '#17365c',
  success: '#4caf50',
  vsCodeDark: '#1e1e1e',
  white: '#fafafa',
} as const;

const fonts = {
  main: "'Open Sans', sans-serif",
} as const;

const fontWeights = {
  bold: 800,
  regular: 400,
  'semi-bold': 600,
} as const;

const DarkTheme = {
  background: colors.black,
  breakpoints,
  colors: {
    ...colors,
    text: colors.white,
  },
  fonts,
  fontWeights,
  svg: colors.white,
  type: 'dark',
};

const LightTheme = {
  background: colors.white,
  breakpoints,
  colors: {
    ...colors,
    text: '#262626',
  },
  fonts,
  fontWeights,
  svg: '#262626',
  type: 'light',
};

type Theme = typeof DarkTheme | typeof LightTheme;
type ThemeProps = {
  background?: string;
  colors?: keyof Theme['colors'] | 'inherit';
  fontWeight?: keyof Theme['fontWeights'];
  svg?: string;
  type?: 'light' | 'dark';
};

export { colors, DarkTheme, GlobalStyle, LightTheme, Theme, ThemeProps };
