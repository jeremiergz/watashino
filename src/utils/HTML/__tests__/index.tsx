import toHast from 'mdast-util-to-hast';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import parse from 'remark-parse';
import { ThemeProvider } from 'styled-components';
import unified from 'unified';
import { renderASTToJSX } from '..';
import { DarkTheme } from '../../../theme';

describe('utils/HTML Test Suite', () => {
  it('matches snapshot', () => {
    const md = [
      '[Gatsby](https://www.gatsbyjs.org)',
      '> blockquote',
      '`inline code`',
      '```\ncode\n```',
      '# h1',
      '## h2',
      '### h3',
      '#### h4',
      '---',
      'text',
    ].join('\n\n');
    const mdast = unified().use(parse).parse(md);
    const jsx = <ThemeProvider theme={DarkTheme}>{renderASTToJSX(toHast(mdast))}</ThemeProvider>;
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
