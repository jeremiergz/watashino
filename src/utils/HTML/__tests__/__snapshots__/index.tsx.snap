// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/HTML Test Suite matches snapshot 1`] = `
<div>
  <p
    className="sc-AxirZ cYqzvF"
    fontSize={20}
  >
    <a
      className="sc-AxiKw sc-AxgMl bFTFRJ"
      href="https://www.gatsbyjs.org"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
      variant="underlined"
    >
      Gatsby
    </a>
  </p>
  

  <blockquote
    className="sc-AxiKw NQHIh"
    color="gray"
    fontStyle="italic"
  >
    <div
      className="sc-AxiKw sc-AxheI ewSbbe"
      height="100%"
    />
    

    <p
      className="sc-AxirZ cYqzvF"
      fontSize={20}
    >
      blockquote
    </p>
    

  </blockquote>
  

  <p
    className="sc-AxirZ cYqzvF"
    fontSize={20}
  >
    <code
      className="sc-AxirZ dNannA"
      color="white"
      display="inline-flex"
      fontSize={16}
    >
      inline code
    </code>
  </p>
  

  <pre>
    <code
      className="sc-AxirZ dNannA"
      color="white"
      display="inline-flex"
      fontSize={16}
    >
      code

    </code>
  </pre>
  

  <span
    className="sc-AxirZ sc-AxhUy gKdEdr"
  >
    h1
  </span>
  

  <span
    className="sc-AxirZ sc-AxhUy hClPml"
  >
    h2
  </span>
  

  <span
    className="sc-AxirZ sc-AxhUy daxbBG"
  >
    h3
  </span>
  

  <span
    className="sc-AxirZ sc-AxhUy lgyYxC"
  >
    h4
  </span>
  

  <hr
    className="sc-AxiKw fgwXnd"
  />
  

  <p
    className="sc-AxirZ cYqzvF"
    fontSize={20}
  >
    text
  </p>
</div>
`;
